common/src/test/kotlin/io/homeassistant/companion/android/database/server/ServerConnectionInfoTest.kt

```diff
@@ -1,335 +1,349 @@
-    }
 
-    @AfterEach
-    fun tearDown() {
-        unmockkAll()
-    }
+    @Nested
+    inner class HasHomeNetworkSetup {
+
+        @Test
+        fun `Given no home network configuration then hasHomeNetworkSetup is false`() {
+            val connection = ServerConnectionInfo(
+                externalUrl = "https://example.com",
+                internalSsids = emptyList(),
+                internalVpn = null,
+                internalEthernet = null,
+            )
 
-    private fun createServerConnectionInfo(
-        externalUrl: String = "https://external.example.com",
-        internalUrl: String? = null,
-        cloudUrl: String? = null,
-        internalSsids: List<String> = emptyList(),
-        internalEthernet: Boolean? = null,
-        internalVpn: Boolean? = null,
-    ): ServerConnectionInfo {
-        return ServerConnectionInfo(
-            externalUrl = externalUrl,
-            internalUrl = internalUrl,
-            cloudUrl = cloudUrl,
-            internalSsids = internalSsids,
-            internalEthernet = internalEthernet,
-            internalVpn = internalVpn,
-        ).apply {
-            this.wifiHelper = this@ServerConnectionInfoTest.wifiHelper
-            this.networkHelper = this@ServerConnectionInfoTest.networkHelper
+            assertFalse(connection.hasHomeNetworkSetup)
         }
-    }
 
-    @Test
-    fun `Given HTTPS URL when checking security status then returns Secure`() {
-        val serverInfo = createServerConnectionInfo()
-        val httpsUrl = "https://example.com"
+        @Test
+        fun `Given internalSsids configured then hasHomeNetworkSetup is true`() {
+            val connection = ServerConnectionInfo(
+                externalUrl = "https://example.com",
+                internalSsids = listOf("HomeWiFi"),
+            )
 
-        val result = serverInfo.currentSecurityStatusForUrl(context, httpsUrl)
+            assertTrue(connection.hasHomeNetworkSetup)
+        }
 
-        assertInstanceOf(SecurityStatus.Secure::class.java, result)
-    }
+        @Test
+        fun `Given internalVpn enabled then hasHomeNetworkSetup is true`() {
+            val connection = ServerConnectionInfo(
+                externalUrl = "https://example.com",
+                internalVpn = true,
+            )
 
-    @Test
-    fun `Given HTTP URL when device is on internal network via ethernet then returns Secure`() {
-        val serverInfo = createServerConnectionInfo(
-            internalUrl = "http://192.168.1.1:8123",
-            internalEthernet = true,
-        )
-        every { networkHelper.isUsingEthernet() } returns true
-        val httpUrl = "http://192.168.1.1:8123"
+            assertTrue(connection.hasHomeNetworkSetup)
+        }
 
-        val result = serverInfo.currentSecurityStatusForUrl(context, httpUrl)
+        @Test
+        fun `Given internalVpn disabled then hasHomeNetworkSetup is false`() {
```

Comment:
Similar to the previous comment I think this one needs to be more complete as it is misleading otherwise

```suggestion
        fun `Given no internalSsids and internalVpn disabled and internalEthernet null then hasHomeNetworkSetup is false`() {
```